consensus/types/src/chain_spec.rs

```diff
@@ -865,6 +865,28 @@ impl ChainSpec {
         )
     }
 
+    /// Returns the slot at which the proposer shuffling was decided.
+    ///
+    /// The block root at this slot can be used to key the proposer shuffling for the given epoch.
+    pub fn proposer_shuffling_decision_slot<E: EthSpec>(&self, epoch: Epoch) -> Slot {
+        if self.fork_name_at_epoch(epoch).fulu_enabled() {
```

Comment:
Should this change be 
```suggestion
        if self.fork_name_at_epoch(epoch + 1).fulu_enabled() {
```
? The first epoch of the fulu fork the proposers are computed at the transition which is decided by the last slot of the previous epoch

Reply:
I think `epoch - 1`, but yes. Working on a PR for this and other correctness fixes now.

Reply:
- https://github.com/sigp/lighthouse/pull/8128